MetaMask/metamask-extension
# Add Custom Network UI

### How To Test Locally
1. Open https://github.com/MetaMask/metamask-extension/blob/eb90cab73c74f91cdedf8d86accd6371e562078a/app/scripts/lib/rpc-method-middleware/handlers/index.js
2. uncomment the commented lines
3. start the extension
4. Make sure you don't have a custom network with `chainId` `0x4d`
5. use the snippet below while on the test-dapp

```js
await ethereum.request({
  method: 'wallet_addEthereumChain',
  params: [{
    chainId: '0x4d',
    rpcUrls: ['https://sokol.poa.network'],
    chainName: 'XDAI (Sokol)',
    nativeCurrency: { symbol: 'XDAI', decimals: 18 },
    blockExplorerUrls: ['https://blockscout.com/poa/sokol']
  }]
})
```
